neat_ml/tests/test_plotting.py

```diff
@@ -0,0 +1,379 @@
+from neat_ml.utils import figure_utils
+
+# TODO: enforce style consistency with ``black`` (issue #11)
+@pytest.fixture(scope="session")
+def synthetic_df() -> pd.DataFrame:
+    """Deterministic (seeded) composition/phase dataframe."""
+    rng = np.random.default_rng(7)
+    x = rng.uniform(0.0, 20.0, 30)
+    y = rng.uniform(0.0, 20.0, 30)
+    phase = (x + y > 20.0).astype(int)
+    return pd.DataFrame({"Sodium Citrate (wt%)": x,
+                         "PEO 8 kg/mol (wt%)": y,
+                         "Phase": phase})
+
+@pytest.fixture(scope="session")
+def baseline_dir() -> Generator[Any, Any, Any]:
+    """
+    Directory that stores the reference (expected) images.
+    """
+    ref = resources.files("neat_ml.tests") / "baseline"
+    with resources.as_file(ref) as path:
+        yield path
+
+def test_plot_gmm_decision_regions_visual_and_logic(
+    tmp_path: Path,
+    baseline_dir: Path,
+    synthetic_df: pd.DataFrame,
+):
+    fig, ax = plt.subplots(figsize=(6, 6), dpi=150)
+    gmm, labels, boundary = figure_utils.plot_gmm_decision_regions(
+        df=synthetic_df,
+        x_col=synthetic_df.columns[0],
+        y_col=synthetic_df.columns[1],
+        phase_col="Phase",
+        ax=ax,
+        xrange=[0, 20],
+        yrange=[0, 20],
+        n_components=2,
+        random_state=42,
+        boundary_color="red",
+        resolution=200,
+        decision_alpha=1,
+        plot_regions=True,
+        region_colors=["lightsteelblue", "aquamarine"],
+    )
+    assert labels.shape == (len(synthetic_df),)
+    assert boundary is None or boundary.shape[1] == 2
+    assert hasattr(gmm, "predict")
+
+    out_png = tmp_path / "gmm_decision_regions.png"
+    fig.savefig(out_png, bbox_inches="tight")
+    plt.close(fig)
+    
+    # for ``compare_images`` tests ``tol`` was loosened from 1e-4 to
+    # 2e-2 to accommodate cross-platform testing as discussed at
+    # https://github.com/lanl/ldrd_neat_ml/pull/1#issuecomment-3463967764
+    result = compare_images(
+        str(baseline_dir / "gmm_decision_regions.png"), 
+        str(out_png), 
+        tol=2e-2)
+    assert result is None
+
+def test_plot_gmm_composition_phase_visual_and_logic(
+    tmp_path: Path,
+    baseline_dir: Path,    
+    synthetic_df: pd.DataFrame,
+):
+    fig, ax = plt.subplots(figsize=(6, 6), dpi=150)
+    figure_utils.plot_gmm_composition_phase(
+        df=synthetic_df,
+        x_col=synthetic_df.columns[0],
+        y_col=synthetic_df.columns[1],
+        phase_col="Phase",
+        ax=ax,
+        point_cmap=["#FF8C00", "dodgerblue"],
+    )
+    scatters = [c for c in ax.collections if np.asarray(c.get_offsets()).size > 0]
+    assert len(scatters) == 2
+
+    out_png = tmp_path / "gmm_composition_phase.png"
+    fig.savefig(out_png, bbox_inches="tight")
+    plt.close(fig)
+
+    result = compare_images(
+        str(baseline_dir / "gmm_composition_phase.png"), 
+        str(out_png), 
+        tol=2e-2)
+    assert result is None
```

Comment:
This test looks identical to two of the cases already covered over in `test_plotters_visual_and_logic()` in the other test module, `test_utils_module.py`.

I can't think of a good reason to duplicate the testing--reason for design decision?